bundles/cxx_remote_services/integration/src/TestExportImportRemoteServiceFactory.cc

```diff
@@ -263,13 +263,16 @@ class CalculatorImportServiceFactory final : public celix::rsa::IImportServiceFa
         auto& cmp = ctx->getDependencyManager()->createComponent(std::make_unique<ImportedCalculator>(logHelper));
         cmp.createServiceDependency<pubsub_publisher>(PUBSUB_PUBLISHER_SERVICE_NAME)
                 .setRequired(true)
+                .setStrategy(DependencyUpdateStrategy::locking)
```

Comment:
Should it also not work stable with suspend strategy?

Reply:
Yes, you are correct is should not matter.

The suspend or locking strategy should not matter because the component should only start when all required  dependencies are available (and all deps are required). 

I though the component maybe was starting ans stopping which resulted in flaky / unstable test result due to timing issue, but this should not be the case. 

The problem for me is that I cannot reproduce this issue on my local machine (ubuntu or mac os), but I will test if the test are still stable if this is reverted to suspend instead of locking.

Reply:
I think I understand why the locking instead of suspend is needed.

The remote pushstream and promise test is done inside a `ctx->useService` call and this call is executed on the Celix event thread. Because pubsub connection are not guaranteed to be setup before using pubsub, it can happen that the first calls of a stream and/or promise fail and this connections cannot be updated because the Celix event thread is busy.

Reply:
I added a retry loop outside of the `ctx->useService` IMO this is cleaner and now the test seems stable.

Note that although I have change/improved some pubsub code, these changes were not the fix for the issue. The fix for #397 was a better test setup.